src/spawn/spawn_server.rs

```diff
@@ -0,0 +1,427 @@
+    // transmute posix_spawn.__actions to own file actions object st. we can access 'private' fields
+    let file_actions = if !file_actions_p.is_null() {
+        let c_file_actions = unsafe { *file_actions_p };
+        Some(unsafe { std::mem::transmute::<posix_spawn_file_actions_t, PosixFileActions>(c_file_actions) })
+    } else {
+        None
+    };
+
+    // adapted from musl-libc: https://github.com/esmil/musl/blob/master/src/process/posix_spawn.c
+    if let Some(file_actions) = file_actions {
+        let mut file_actions_op_p: *const Fdop = unsafe { std::mem::transmute::<*mut libc::c_int, *const Fdop>(file_actions.__actions) };
+        while !file_actions_op_p.is_null() {
+            let op = unsafe { *file_actions_op_p };
+            
+            match op.cmd {
+                // if FD is not 0,1,2: no action required; simply don't get the FD from the client
+                // otherwise: close the FD
+                FDOP_CLOSE => {
+                    match op.fd {
+                        libc::STDIN_FILENO | libc::STDOUT_FILENO | libc::STDERR_FILENO => {
+                            nix::unistd::close(op.fd).unwrap();
+                        }
+                        _ => (),
+                    }
+                },
+                // no action required; `posix_spawn` can open any path on its own, without
+                // any further info required from the client
+                FDOP_OPEN => {},
+                // get the client fd, then dup2 it
+                FDOP_DUP2 => {
+                    let client_fd: i32 = unsafe {
+                        libc::syscall(libc::SYS_pidfd_getfd, client_pid_fd, op.srcfd, 0)
+                            .try_into()
+                            .unwrap()
+                    };
+
+                    if client_fd < 0 {
+                        panic!("pidfd_getfd: {}", std::io::Error::last_os_error());
+                    }
+
+                    let _ = nix::unistd::dup2(client_fd, op.fd).unwrap();
+
+                    // close no longer needed fd to avoid collisions with further `dup`s
+                    nix::unistd::close(client_fd).unwrap();
+                },
+                n => panic!("Got invalid fdop cmd: {n}"),
+            }
+
+            file_actions_op_p = unsafe {
+                let faop = *file_actions_op_p;
+                faop.next
+            };
+        }
+    }
+
+    // let c_str = CString::new("TARGET PID: %d\n").unwrap();
+    // let c_str_2 = CString::new("TARGET PPID: %d\n").unwrap();
+    // unsafe { libc::printf(c_str.as_ptr(), libc::getpid()); };
+    // unsafe { libc::printf(c_str_2.as_ptr(), libc::getppid()); };
+
+    // unsafe { libc::execvpe(executable, ptr_array, envp.as_ptr()) };
+    let spawn_func = if use_path {
+        libc::posix_spawn
+    } else {
+        libc::posix_spawnp
+    };
+
+    
+    if unsafe { spawn_func(
+        pid, executable, file_actions_p, spawn_attrs, argv, envp
+    ) } < 0 {
```

Comment:
```suggestion
    ) } != 0 {
```

The man page says that 0 is returned on success and doesn't specify the range of values in the case of an error.

During my testing it returned positive values when it failed (for some reason).